libtransact/src/scheduler/multi/mod.rs

```diff
@@ -228,16 +229,16 @@ impl MultiSchedulerFactory {
 }
 
 impl SchedulerFactory for MultiSchedulerFactory {
-    fn create_scheduler(&mut self, state_id: String) -> Result<Box<dyn Scheduler>, SchedulerError> {
+    fn create_scheduler(&self, state_id: String) -> Result<Box<dyn Scheduler>, SchedulerError> {
         let schedulers = self
             .sub_scheduler_factories
-            .iter_mut()
+            .iter()
             .flat_map(|(factory, num)| {
                 let state_id = state_id.clone();
                 (0..*num).map(move |_| factory.create_scheduler(state_id.clone()))
             })
             .collect::<Result<_, _>>()?;
-        MultiScheduler::new(schedulers, &mut *self.sub_scheduler_handler)
+        MultiScheduler::new(schedulers, &mut **self.sub_scheduler_handler.borrow_mut())
```

Comment:
Won't this byte you if this is called more than once?

Reply:
Updated to use `try_borrow_mut` to avoid the possibility of a panic.